extensions/data-workspace/src/main.ts

```diff
@@ -41,5 +47,21 @@ export function activate(context: vscode.ExtensionContext): Promise<IExtension>
 	return Promise.resolve(dataWorkspaceExtension);
 }
 
+function setTreeDataProvider(context: vscode.ExtensionContext, workspaceService: IWorkspaceService, treeDataProvider: WorkspaceTreeDataProvider): void {
+	const isProjectProviderAvailable = workspaceService.isProjectProviderAvailable;
+	vscode.commands.executeCommand('setContext', 'isProjectProviderAvailable', isProjectProviderAvailable);
+	if (isProjectProviderAvailable) {
+		if (treeDataProviderDisposable) {
+			treeDataProviderDisposable.dispose();
```

Comment:
why does this need to be disposed and recreated instead of skipping creating it if treeDataProviderDisposable isn't undefined when a new extension is added?

Reply:
actually, this is no longer needed, it is required when we want to show the view when no provider is installed. I have simplified this.